### PR TITLE
[nmi_gen] Add a wrapper peripheral for the escalation receivers for testing purposes 

### PR DESCRIPTION
**Pull request by msfschaffner**
Carried over from #674.

This PR adds `nmi_gen`, a simple wrapper peripheral that receives the escalation signals from the alert handler and converts them to interrupts for the core. These are currently not true non-maskable interrupts. The module is only used for testing the alert handler mechanism at top level, and will be adapted and/or replaced in the future.

Signed-off-by: Michael Schaffner <msf@google.com>

### CODE REVIEW

File: hw/ip/nmi_gen/data/nmi_gen.hjson [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+//
+# NMI_GEN register template
+{
+  name: "NMI_GEN",
+  clock_primary: "clk_i",
+  bus_device: "tlul",
+  regwidth: "32",
+  param_list: [
```
Comment (sjgitty) [inline hw/ip/nmi_gen/data/nmi_gen.hjson]: is `param_list` needed if it is an empty list?

File: hw/ip/nmi_gen/doc/index.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+This document specifies the functionality of the non-maskable interrupt generator (`nmi_gen`) peripheral.
+This module conforms to the [OpenTitan guideline for peripheral device functionality.]({{< relref "doc/rm/comportability_specification" >}}).
+See that document for integration overview within the broader OpenTitan top level system.
+The module provides a mechanism to test the alert handler escalation signals (see [alert handler spec]({{< ref "hw/ip/alert_handler/doc" >}})), and will be modified or even replaced in the near future.
+
+
+## Features
+
+- Contains four escalation receivers to receive escalation signals from the alert handler
+
```
Comment (sjgitty) [inline hw/ip/nmi_gen/doc/index.md]: probably don't need the empty line here.

File: hw/ip/nmi_gen/doc/index.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+- Produces four regular interrupts that are derived from the alert handler escalation signals
+
+## Description
+
+The NMI generator module is a simple wrapper module that instantiates four escalation receivers (which are described in more detail in the alert handler spec.
```
Comment (sjgitty) [inline hw/ip/nmi_gen/doc/index.md]: closed paren) needed

**Comment by msfschaffner** [comment]
I added the .prj. file for consistency. We could just keep the module below D3 until we remove it again...

File: hw/ip/nmi_gen/doc/index.md (hunk 3 above)
Comment (msfschaffner) [reply hw/ip/nmi_gen/doc/index.md]: fixed

File: hw/ip/nmi_gen/doc/index.md (hunk 2 above)
Comment (msfschaffner) [reply hw/ip/nmi_gen/doc/index.md]: fixed

File: hw/ip/nmi_gen/data/nmi_gen.hjson (hunk 1 above)
Comment (msfschaffner) [reply hw/ip/nmi_gen/data/nmi_gen.hjson]: removed

**Comment by msfschaffner** [comment]
Fixed and rebased.

**Comment by tjaychen** [comment]
hmmm...  i actually thought about the escalations a little differently.
Ie, since alert_handler shoots out 4 escalation signals, they would actually go to different places in the design.  One would perhaps go to the ibex wrapper, and be used for the generation of the nmi, one would go to the power sequencer in preparation for a reset, and perhaps another would go to the not-yet-there life cycle manager for a different type of response.

So I don't really envision it going to another centralized location with signals that are again broadcast out.  This is mainly for a two reasons..

1. if this nmi_gen module's outputs will travel long distances to other destinations, then we just lost some of the differential properties and heartbeat protections.  It makes more sense for the responder modules to directly receive them. 

2. if we sent them to a centralized location, it would make just as much as sense to keep it in alert_handler directly, since most of the properties would be the same.

Having said all that, I think it's fine for this to go in, but we should make it really clear, either in an issue (and it's already there in the docs) that will very likely be replaced by a different scheme and this is primarily for testing. 

@cdgori , could you comment a bit on whether this aligns with your thinking?

Sorry I know a lot of this is a result of there being no top level clarity on how everything should fit together.. that is something that we collectively should work on.

**Comment by msfschaffner** [comment]
@tjaychen I completely agree with you. This is by no means meant to last, but I needed a module where I can connect these escalation signals to, and at the moment, these escalation endpoints simply do not exist yet. Hence I decided a wrapper module such as `nmi_gen` could bridge the gap for testing purposes until we actually have endpoints for the escalation signals. Sorry if I did not make that clear in the PR description. I opened an issue to track this.

**Pull request merged**